### PR TITLE
DOC Integration + Mark

### PR DESCRIPTION
**Pull request by gabrielsaruhashi**
Remaining DOC endpoints + file upload

Closes issue #45 
Closes issue

### CODE REVIEW

File: src/authentication/authentication.service.ts [hunk 1]
```diff
@@ -79,10 +79,7 @@ export default class AuthenticationService {
       );
       return user;
     } catch (error) {
-      throw new HttpException(
-        'Wrong credentials provided',
-        HttpStatus.BAD_REQUEST,
-      );
+      throw new HttpException(JSON.stringify(error), HttpStatus.BAD_REQUEST);
```
Comment (lspdrz) [inline src/authentication/authentication.service.ts:82]: Nice!!! Overdue

File: src/facilities/facilities.controller.ts [hunk 2]
```diff
@@ -75,21 +77,37 @@ export default class FacilitiesController {
...
       total,
-      page: query.page,
-      limit: query.limit,
+      page,
+      limit,
     };
   }
 
+  @Get(':facilityId/calls/:callId/callMessages')
+  @UseGuards(StaffGuard)
```
Comment (lspdrz) [inline src/facilities/facilities.controller.ts:100]: have you merged in the latest master? I think this StaffGuard is at the class level now

File: src/facilities/services/groups.service.ts [hunk 3]
```diff
@@ -33,6 +31,17 @@ export default class GroupsService {
     const kiosks = parentGroups
       .map((group) => group.kiosks)
       .reduce((prev, curr) => [...prev, ...curr], []);
+
     return kiosks;
   }
+
```
Comment (lspdrz) [inline src/facilities/services/groups.service.ts:37]: Wow this is awesome and so easy!!!!!

How expensive is that findTrees() query??

File: src/inmates/entities/inmate.entity.ts [hunk 4]
```diff
@@ -22,7 +22,9 @@ class Inmate extends BaseEntity {
   @Column({ nullable: true })
   personId: number;
 
-  @OneToOne(() => Person, (person: Person) => person.user)
+  @OneToOne(() => Person, (person: Person) => person.user, {
+    cascade: ['update'],
+  })
```
Comment (lspdrz) [inline src/inmates/entities/inmate.entity.ts:27]: What is this for?

**Review by lspdrz — COMMENTED** [review]
Looking great!! Just a few questions

File: src/inmates/entities/inmate.entity.ts (hunk 4 above)
Comment (gabrielsaruhashi) [reply src/inmates/entities/inmate.entity.ts:27]: image
plain cascade seems too op

File: src/facilities/services/groups.service.ts (hunk 3 above)
Comment (gabrielsaruhashi) [reply src/facilities/services/groups.service.ts:37]: not sure tbh, but closure tables are optimized for both reads and writes, so I guess it's the most efficient way to do it

File: src/facilities/facilities.controller.ts (hunk 2 above)
Comment (gabrielsaruhashi) [reply src/facilities/facilities.controller.ts:100]: I'll fix the merge conflict and change this

**Review by lspdrz — APPROVED** [review]

**Pull request merged**